### PR TITLE
Port `:vertica` driver to Honey SQL 2

### LINKED ISSUE #28166: Migrate `:vertica` driver to Honey SQL 2

### PR DESCRIPTION
**Pull request by camsaul**
Resolves #28166

### CODE REVIEW

File: src/metabase/driver/sql/query_processor.clj [hunk 1]
```diff
@@ -182,7 +182,7 @@
 
 (defn- format-compiled
   [_compiled [honeysql-expr]]
-  (sql/format-expr honeysql-expr {:inline true}))
+  (sql/format-expr honeysql-expr {:nested true}))
```
Comment (camsaul) [inline src/metabase/driver/sql/query_processor.clj:185]: This should have been `{:nested true}` all along

File: src/metabase/driver/sql_jdbc/execute.clj [hunk 2]
```diff
@@ -348,7 +348,16 @@
...
   ^PreparedStatement [driver conn sql params canceled-chan]
-  (doto (prepared-statement driver conn sql params)
+  ;; sometimes preparing the statement fails, usually if the SQL syntax is invalid.
+  (doto (try
+          (prepared-statement driver conn sql params)
+          (catch Throwable e
+            (throw (ex-info (tru "Error preparing statement: {0}" (ex-message e))
+                            {:driver driver
+                             :type   qp.error-type/driver
+                             :sql    (str/split-lines (mdb.query/format-sql sql driver))
```
Comment (metamben) [inline src/metabase/driver/sql_jdbc/execute.clj:358]: Splitting is somewhat surprising.

**Review by metamben — APPROVED** [review]
LGTM

File: src/metabase/driver/sql_jdbc/execute.clj (hunk 2 above)
Comment (camsaul) [reply src/metabase/driver/sql_jdbc/execute.clj:358]: I've found that it's a lot easier to read when it gets printed out than something like `"SELECT\x  x,\n  y,..."`

**Pull request merged**